modules/celtuce-core/src/celtuce/connector.clj

```diff
@@ -278,8 +278,12 @@
     {codec (nippy-codec)
      client-options {}}}]
   (let [redis-client (if (nil? client-resources)
-                       (RedisClient/create redis-uri)
-                       (RedisClient/create client-resources redis-uri))
+                       (if (instance? RedisURI redis-uri)
```

Comment:
How about something like
(cond
  (and (nil? client-resources) (instance? RedisURI redis-uri)
   ;; etc ...
   )

Reply:
Sure. I’ll make that change Monday. What do you think of the RedisURI builder function I posted? Worth cleaning that up and adding that as a utility function?

Reply:
Yes it looks fine too ! Let's add it as a utility function (through your PR).